pkg/cmd/dcos.go

```diff
@@ -226,22 +225,13 @@ func invokePlugin(ctx api.Context, cmd plugin.Command, args []string) error {
 			execCmd.Env = append(execCmd.Env, "DCOS_TLS_CA_PATH="+cluster.TLS().RootCAsPath)
 		}
 	}
+
 	err = execCmd.Run()
 	if err != nil {
 		// Because we're silencing errors through Cobra, we need to print this separately.
 		ctx.Logger().Debug(err)
-
-		// When the plugin command exits with a non-zero code, the main CLI process
-		// exit code should be the same.
-		//
-		// see https://jira.mesosphere.com/browse/DCOS_OSS-4399
-		if exiterr, ok := err.(*exec.ExitError); ok {
-			if status, ok := exiterr.Sys().(syscall.WaitStatus); ok {
-				// TODO: have a more generic error handling system
-				// where errors can be associated with an exit code.
-				os.Exit(status.ExitStatus())
-			}
-		}
+		os.Exit(execCmd.ProcessState.ExitCode())
```

Comment:
according to https://golang.org/pkg/os/#ProcessState.ExitCode, this might return -1 which is not a valid exit code. We should only exit if the process terminated with a code > 0.